utilities/include/sparse.h

```diff
@@ -161,6 +178,10 @@ class SparseIndices {
   }
 
  private:
-  VecDH<int> p, q;
+  VecDH<uninitialized_int> p, q;
+  VecDH<int>& GetP() { return reinterpret_cast<VecDH<int>&>(p); }
+  VecDH<int>& GetQ() { return reinterpret_cast<VecDH<int>&>(q); }
+  const VecDH<int>& GetP() const { return reinterpret_cast<const VecDH<int>&>(p); }
+  const VecDH<int>& GetQ() const { return reinterpret_cast<const VecDH<int>&>(q); }
```

Comment:
I guess these casts what the compiler is angry about?

Reply:
Yes it seems so, interestingly clang++ 13 did not complain about it (I use clang because my openmp is the one from LLVM).
I was looking at other ways to do this, like union/overriding vector allocator, but it seems that I need to cast the vector type anyway. Not sure if there is any idiomatic way of doing this in C++.

Reply:
It seems that the slowness due to initialization is caused by a large initial size for sparse indices. I've reduced the initial size and fixed the growth rate of it, so this change can be dropped.